needle-slack-connector/src/app/_components/providers/SlackResourcesProvider.tsx

```diff
@@ -0,0 +1,144 @@
+"use client";
+
+import { createContext, useContext, useState, useEffect } from "react";
+import {
+  type SlackMessage,
+  type SlackWorkspace,
+} from "~/models/connectors-models";
+import { type TimezoneInfo } from "~/server/slack/service";
+import { type SlackChannel } from "~/server/db/schema"; // Use SlackChannel type from schema
+
+import { api } from "~/trpc/react";
+import { skipToken } from "@tanstack/react-query";
+
+type SlackResourcesContextType = {
+  selectedChannels: SlackChannel[]; // Use SlackChannel type
+  setSelectedChannels: (channels: SlackChannel[]) => void;
+  channels: SlackChannel[] | undefined;
+  messages: SlackMessage[] | undefined;
+  workspace: SlackWorkspace | undefined;
+  userTimezone: TimezoneInfo | undefined;
+  isLoading: boolean;
+};
+
+const SlackResourcesContext = createContext<SlackResourcesContextType | null>(
+  null,
+);
+
+interface SlackResourcesProviderProps {
+  children: React.ReactNode;
+  credentials: string;
+  userId: string;
+}
+
+export function SlackResourcesProvider({
+  children,
+  credentials,
+  userId,
+}: SlackResourcesProviderProps) {
+  const [selectedChannels, setSelectedChannels] = useState<SlackChannel[]>([]);
+
+  const { data: workspace, isLoading: isWorkspaceLoading } =
+    api.connectors.getWorkspaces.useQuery({
+      accessToken: credentials,
+    });
+
+  const { data: channels, isLoading: isChannelsLoading } =
+    api.connectors.getChannels.useQuery({
+      accessToken: credentials,
+    });
+
+  const { data: userTimezone, isLoading: isTimezoneLoading } =
+    api.connectors.getUserTimezone.useQuery({
+      accessToken: credentials,
+      userId,
+    }) as { data: TimezoneInfo | undefined; isLoading: boolean };
+
+  const { data: messages, isLoading: isMessagesLoading } =
+    api.connectors.getMessages.useQuery(
+      selectedChannels.length > 0
+        ? {
+            accessToken: credentials,
+            channelIds: selectedChannels.map((channel) => channel.id),
+          }
+        : skipToken,
```

Comment:
nit: if we conditionally enable, then do we need `skipToken`?

Reply:
Correct, we don't need it I will remove